glue/sample/src/sinter/_decoding_mwpf.py

```diff
@@ -0,0 +1,262 @@
+import math
+import pathlib
+from typing import Callable, List, TYPE_CHECKING, Tuple
+
+import numpy as np
+import stim
+
+from sinter._decoding_decoder_class import Decoder, CompiledDecoder
+
+if TYPE_CHECKING:
+    import mwpf
+
+DEFAULT_TIMEOUT: float = 10.0  # decoder timeout in seconds
```

Comment:
Whoah whoah whoah, this timeout is a major problem.

Sinter is a bulk sampling library. The workers collecting samples could be executing under widely different loads, or on machines of widely different speeds. Specifying a timeout in seconds means that you could get different statistics from running on different machines, or at different times, or with different numbers of workers. It completely breaks the assumption that the collected samples can be merged while assuming that they are from the same distribution.

This needs to be replaced with something like a maximum number of steps, or a chance-of-giving-up-per-step, or some other quantity that will give the same distribution today as it would give tomorrow.

Back in the old days there were some compilers that used timeouts to determine when to stop optimizing a program. They were a disaster, because the performance of the resulting program would change from run to run.

Reply:
Yeah, this makes a lot of sense! Thanks for the feedback and I'll add mechanism to deterministically limit the runtime. 

One question is whether it's possible to pass a configuration object to the decoder construction? I don't see such configuration in PyMatching or other decoders so I'm just wondering what's your suggestion of doing this, especially when everything is handled by sinter.Task. It could be useful for other decoders as well.

Reply:
I have updated the implementations to avoid timeout. Now the class corresponds to a decoder without any timeout, so it's probably running too long for some syndromes. We'll add some mechanism later, but the current implementation seems to work, with some benchmark results commented below.